### ISSUE TITLE
Static tests: forbid 'or' instead of '||'

### ISSUE THREAD

**Issue by novikor** [magento/magento2#21062]
### Summary (*)

At the moment, static tests allow to use `or` operator which, however, is bad coding practice.
There was a PR related to this problem: https://github.com/magento/magento2/pull/20628

I believe ot would be nice to include this check in the static tests.
### Proposed solution
Update static tests rules to cover this case

**Comment by magento-engcom-team** [comment]
Hi @novikor. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@novikor do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @novikor   thank you for you report, It will be processed faster if you move this to https://github.com/magento/community-features

**Comment by novikor** [comment]
@engcom-backlog-nazar , OK, I`ll do it.

However, I can try to create a PR to this repository with this improvement by myself. Am I right?

**Comment by orlangur** [comment]
@novikor surely you can ;)

**Comment by ghost** [comment]
@novikor Yes feel fre to create a Pull Request.

**Comment by orlangur** [comment]
@novikor the best way to introduce such check would be to implement a long-awaited static test involving `php-cs-fixer` as it has a rule
```
logical_operators [@PhpCsFixer:risky]

Use && and || logical operators instead of and and or.

Risky rule: risky, because you must double-check if using and/or with lower precedence was intentional.
```

**Comment by orlangur** [comment]
Oh, there is existing `phpcs` sniff actually, thanks @lenaorobei for finding this out.

Closing in favor of https://github.com/magento/magento-coding-standard/issues/23.

@novikor please consider `php-cs-fixer` integration anyway ;)

**Comment by novikor** [comment]
@orlangur , I have found a solution how to apply this rule to phpcs and static tests.
Ruleset.xml is present in `magento/magento-coding-standard`, but [.php_cs.dist](https://github.com/magento/magento2/blob/2.3-develop/.php_cs.dist) is not.

May I create a PR in this repository?

**Comment by orlangur** [comment]
@novikor sure thing! Are you going to implement a proper `php-cs-fixer` integration?

**Comment by novikor** [comment]
> @novikor sure thing! Are you going to implement a proper `php-cs-fixer` integration?

Yes, I will try

**Comment by magento-engcom-team** [comment]
Hi @novikor. Thank you for your report. 
The issue has been fixed in magento/magento2#21275 by @novikor in 2.3-develop branch
Related commit(s):
 - [23f415a5d7f7f032a33f75ced47a76b4b422ef7e](https://github.com/magento/magento2/commit/23f415a5d7f7f032a33f75ced47a76b4b422ef7e)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @novikor. Thank you for your report. 
The issue has been fixed in magento/magento2#21543 by @novikor in 2.2-develop branch
Related commit(s):
 - [9f0ef596df5e7e97d4057e4a84eaf18f08cbb5e0](https://github.com/magento/magento2/commit/9f0ef596df5e7e97d4057e4a84eaf18f08cbb5e0)

The fix will be available with the upcoming 2.2.9 release.
